name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/PortfolioReportNet.java

```diff
@@ -209,30 +211,28 @@ else if (!symbol.equals(property.getValue()))
 
     public List<ResultItem> search(String query, SecuritySearchProvider.Type type) throws IOException
     {
-        try
+        try (HttpClient httpClient = new HttpClient())
         {
-            URIBuilder uriBuilder = new URIBuilder().setScheme("https").setHost(HOST) //$NON-NLS-1$
-                            .setPath("/api/securities/search/" + query); //$NON-NLS-1$
+
+            httpClient.setURIScheme("https"); //$NON-NLS-1$
+            httpClient.setURIHost(HOST);
+            httpClient.setURIPath("/api/securities/search/" + query); //$NON-NLS-1$
+            httpClient.setURIBuilder();
 
             if (type != null)
             {
                 if (type == SecuritySearchProvider.Type.SHARE)
-                    uriBuilder.addParameter("type", TYPE_SHARE); //$NON-NLS-1$
+                    httpClient.getURIBuilder().addParameter("type", TYPE_SHARE); //$NON-NLS-1$
                 else if (type == SecuritySearchProvider.Type.BOND)
-                    uriBuilder.addParameter("type", TYPE_BOND); //$NON-NLS-1$
+                    httpClient.getURIBuilder().addParameter("type", TYPE_BOND); //$NON-NLS-1$
             }
 
-            URL searchUrl = uriBuilder.build().toURL();
-
-            URLConnection con = searchUrl.openConnection();
-            con.setConnectTimeout(1000);
-            con.setReadTimeout(5000);
-
+            InputStream con = new ByteArrayInputStream(httpClient.requestData().getBytes());
```

Comment:
Ich glaube das ist unnötig, weil etwas weiter unten der InputStream wieder in einen String gelesen wird. Da könnte man sich das "new Scanner..." sparen.